pkg/controllers/termination/terminate.go

```diff
@@ -124,6 +124,7 @@ func (t *Terminator) getEvictablePods(pods []*v1.Pod) []*v1.Pod {
 		if pod.IsOwnedByNode(p) {
 			continue
 		}
+		logging.FromContext(ctx).Debugf("pod %s marked as evictable", p.Name)
```

Comment:
Instead of logging here, can we log where we actually send the eviction requests?

Reply:
Yes, but which pods should we be logging in that case?
Given the block of code 
```go
func (t *Terminator) evict(pods []*v1.Pod) {
	// 1. Prioritize noncritical pods https://kubernetes.io/docs/concepts/architecture/nodes/#graceful-node-shutdown
	critical := []*v1.Pod{}
	nonCritical := []*v1.Pod{}
	for _, p := range pods {
		if !p.DeletionTimestamp.IsZero() {
			continue
		}
		if p.Spec.PriorityClassName == "system-cluster-critical" || p.Spec.PriorityClassName == "system-node-critical" {
			critical = append(critical, p)
		} else {
			nonCritical = append(nonCritical, p)
		}
	}
	// 2. Evict critical pods if all noncritical are evicted
	if len(nonCritical) == 0 {
		t.EvictionQueue.Add(critical)
	} else {
		t.EvictionQueue.Add(nonCritical)
	}
}
```
We can either:

- log all pods regardless of if conditions
- log only pods which are scheduled for the actual deletion (which would require additional loops to fetch them)

I remind you that the scope of the patch is to get a clear picture of what's preventing us from destroying the node (well one of the main reasons).

Reply:
What do you think about adding the log line as the first line inside `t.EvictionQueue.Add(ctx, pod)`. This will make sure we don't double count (which I think the current impl does). 

Do you mind adding the logging that this produces in the description of the PR?

Reply:
Sorry for the bad advise -- it looks like we already have this log
```
karpenter-controller-556bddb76-ng7tt manager 2022-02-03T19:32:22.166Z	DEBUG	controller	Evicted pod default/inflate-68785b889b-9s89h	{"commit": "955ed8c"}
```

Reply:
@ellistarn log line you've mentioned is coming from here:
```go
if e.evict(ctx, nn) {
			logging.FromContext(ctx).Debugf("Evicted pod %s", nn.String())
			e.RateLimitingInterface.Forget(nn)
			e.Set.Remove(nn)
			e.RateLimitingInterface.Done(nn)
			continue
		}
```
so we would only have this if 

1. we are actually able to remove the pod
2. the queue is working. 

Adding log in queue.add works for me, something like 
```
func (e *EvictionQueue) Add(pods []*v1.Pod) {
	for _, pod := range pods {
                logging.FromContext(ctx).Debugf("adding pod %s to the eviction queue ", pod.Name())
		if nn := client.ObjectKeyFromObject(pod); !e.Set.Contains(nn) {
			e.Set.Add(nn)
			e.RateLimitingInterface.Add(nn)
		}
	}
}
```

As for amount of logging, these are stdout logs produced only when we set the proper loglevel, so amount of noise during normal operation should not be an issue?